src/ec/suite_b/ops/ops.rs

```diff
@@ -215,11 +210,9 @@ impl CommonOps {
 
     pub fn elem_verify_is_not_zero(&self, a: &ElemUnreduced)
                                    -> Result<(), error::Unspecified> {
-        match unsafe {
-            GFp_constant_time_limbs_are_zero(a.limbs.as_ptr(), self.num_limbs)
-        } {
-            0 => Ok(()),
-            _ => Err(error::Unspecified),
+        match limbs_are_zero_constant_time(&a.limbs[..self.num_limbs]) {
+            false => Ok(()),
+            true => Err(error::Unspecified),
```

Comment:
we have a function `GFp_constant_time_limbs_are_zero` that returns `(Limb)-1` for TRUE and `0` for FALSE. `limbs_are_zero_constant_time` maps `(Limb)-1` => `true` and `0` => `false`. Then we map `true` => `Ok(())` and `false => `Err(error::Unspecified)`. This seems quite circuitous to me. I'll try to write a follow-up commit to this that creates a `BooleanMask` enum type that can be used to remove at least one layer of mapping here.